tips/TIP-0040/tip-0040.md

```diff
@@ -0,0 +1,769 @@
+        <td><b>Type</b></td>
+        <td><b>Description</b></td>
+    </tr>
+    <tr>
+        <td>Epoch Index</td>
+        <td>uint64</td>
+        <td>
+            The epoch index for which the profit margin is tracked.
+        </td>
+    </tr>
+    <tr>
+        <td>Profit Margin</td>
+        <td>uint8</td>
+        <td>
+            An integer representing the epoch profit margin, scaled to `2^ProfitMarginExponent`.
+        </td>
+    </tr>
+</table>
+
+### Epoch Rewards
+
+The rewards of an epoch are tracked by keeping a map with `Validator ID`s as keys, of type `ByteArray[32]`, and the
+values being a list of objects consisting of these fields:
+
+<table>
+    <tr>
+        <td><b>Name</b></td>
+        <td><b>Type</b></td>
+        <td><b>Description</b></td>
+    </tr>
+    <tr>
+        <td>Epoch Index</td>
+        <td><code>uint64</code></td>
+        <td>The epoch index for which the rewards can be claimed.</td>
+    </tr>
+    <tr>
+        <td>Pool Stake</td>
+        <td><code>uint64</code></td>
+        <td>The total amount of IOTA coins staked by the validator and all its delegators.</td>
+    </tr>
+    <tr>
+        <td>Pool Rewards</td>
+        <td><code>uint64</code></td>
+        <td>The total amount of rewards the validator pool received.</td>
+    </tr>
+</table>
+
+The next section defines how these values can be obtained from the slot-level data, after the commitment of the last
+slot of that epoch.
+
+### Slot-Epoch Conversion of the Rewards
+
+The following section specifies how to convert the slot-level data into epoch-level data.
+
+#### Parameters needed for calculations:
+
+- `Bootstrapping Duration` = 1154 (approximately 3 years)
+- `Rewards Mana Share Coefficient` = 2 (relative to the term $\theta/(1-\theta)$ from the Whitepaper, with $\theta =
+  2/3$)
+- `Final Reward` =
+  `(Total Supply * Rewards Mana Share Coefficient * generationRate) >> (generationRateExponent - slotsPerEpochExponent)`
+  (constant reward per epoch after the bootstrapping phase)
+- `Decay Balancing Constant Coefficient` = 8
+- `Decay Balancing Constant =` (integer approximation of $2^{\text{Decay Balancing Constant Coefficient}}\frac{e
+  \Delta}{T(1-\exp{(-\beta \Delta)})}$, for $Delta =$ epoch duration in years, $T=$ `Bootstrapping Duration`, and
+  $\beta=1/3$). For `Decay Balancing Constant Coefficient` = 8, `Decay Balancing Constant` = 696.
+- `Initial Reward` = `(Final Reward * Decay Balancing Constant) >> Decay Balancing Constant Coefficient` (parameter to
+  calculate the reward per epoch in the bootstrapping phase)
+- `Pool Coefficient Exponent` = 11
+
+#### Input values
+
+- Let `Validator Stake(i)` be the amount of IOTA coins staked by the validator with Account ID `i`, i.e. the
+  `Staked Amount` in the _Staking Feature_ of the validator at the end of the registration slot.
+- Let `Pool Stake` be the total amount of delegated and staked IOTA coins by the validator pool which was determined at
```

Comment:
Shouldn't `Pool Stake` be `Pool Stake(i)` just like `Validator Stake(i)` as it corresponds to a specific validator and each of its delegators?

Reply:
Yeah, this TIP needs polish in those formulas!